src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,37 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int MIN_AGE = 35;
+    private static final int LIVING_PERIOD = 10;
+
+    @Override
+    public boolean test(Candidate candidate) {
+        return candidate != null
+                && candidate.getAge() >= MIN_AGE
+                && candidate.isAllowedToVote()
+                && "Ukrainian".equals(candidate.getNationality())
+                && hasLivedInUkraineForAtLeast10Years(candidate.getPeriodsInUkr());
+    }
+
+    public boolean hasLivedInUkraineForAtLeast10Years(String periodsInUkr) {
```

Comment:
let's improve this method
```suggestion
    private boolean isValidByResidenceInUkraine(Candidate candidate) {
        String[] yearsInUkraine = candidate.getPeriodsInUkr().split("\\D");
        return Integer.parseInt(yearsInUkraine[INDEX_LEAVE_TO])
                - Integer.parseInt(yearsInUkraine[INDEX_LEAVE_FROM]) >= LIVING_PERIOD;
    }
```